Extensions/IISWebAppDeploy/Src/Tasks/IISWebAppDeploy/IISWebAppDeployV2/MsDeployOnTargetMachines.ps1

```diff
@@ -0,0 +1,271 @@
+﻿Write-Verbose "Entering script MsDeployOnTargetMachines.ps1"
+$MsDeployInstallPathRegKey = "HKLM:\SOFTWARE\Microsoft\IIS Extensions\MSDeploy"
+
+function Run-Command
+{
+    param(
+        [string]$command,
+        [bool] $failOnErr = $true
+    )
+
+    $ErrorActionPreference = 'Continue'
+
+    if( $psversiontable.PSVersion.Major -le 4)
+    {        
+        $result = cmd.exe /c "`"$command`""
+    }
+    else
+    {
+        $result = cmd.exe /c "$command"
+    }
+    
+    $ErrorActionPreference = 'Stop'
+
+    if($failOnErr -and $LASTEXITCODE -ne 0)
+    {
+        throw $result
+    }
+    
+    return $result
+}
+
+function Get-MsDeployLocation
+{
+    param(
+    [Parameter(Mandatory=$true)]
+    [string]$regKeyPath
+    )
+    
+    if( -not (Test-Path -Path $regKeyPath))
+    {
+        throw (Get-VstsLocString -Key "MsDeployNotFoundError") 
```

Comment:
This script will be executed on the remote machine. `VSTS task sdk` won't be available there. Hence `Get-VstsLocString` won't work.

Reply:
It is done